### PR TITLE
Add multi-vm tests for Orderbook

### PR DESCRIPTION
**Pull request by shubhamgoyal42**
## Why this should be merged
Adds tests where multiple instances of VM simulate multiple nodes in the network.

## How this was tested
```
go test -v -timeout 30s -run ^TestHubbleLogs$ github.com/ava-labs/subnet-evm/plugin/evm
```

### CODE REVIEW

File: plugin/evm/orderbook_test.go [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+
+// vm1 proposes block A containing order 1
+// block A is accepted
+// vm1 proposes block B containing order 2
+// vm1 and vm2 set preference to block B
+// vm2 proposes block C containing order 2 & order 3
+// vm1 and vm2 set preference to block C
+// vm2 proposes block D containing matching tx of order 1 and 2
+// vm1 and vm2 set preference to block D
+// reorg happens when vm1 accepts block D
```
Comment (atvanguard) [inline plugin/evm/orderbook_test.go:90]: if vm1 has already set preference to block c (line 87), why would a reorg accur when D is accepted (line 90)?

File: plugin/evm/orderbook_test.go [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+
+	vm1BlockC := parseBlock(t, vm1, vm2BlockC)
+	setPreference(t, vm1BlockC, vm1)
+	accept(t, vm1BlockC)
+	accept(t, vm2BlockC)
+
+	detail1 := vm1.limitOrderProcesser.GetOrderBookAPI().GetDetailedOrderBookData(context.Background())
+	detail2 := vm2.limitOrderProcesser.GetOrderBookAPI().GetDetailedOrderBookData(context.Background())
+	t.Logf("VM1 Orders: %+v", detail1)
+	t.Logf("VM2 Orders: %+v", detail2)
```
Comment (atvanguard) [inline plugin/evm/orderbook_test.go:134]: shouldn't we have an assertion on `detail1` and `detail2`?
What assertions fail when @fdgod09's last change is commented out?

File: plugin/evm/orderbook_test.go [hunk 3]
```diff
@@ -0,0 +1,262 @@
...
...e":"uint256","name":"","type":"uint256"},{"internalType":"address","name":"","type":"address"}],"name":"positions","outputs":[{"internalType":"int256","name":"size","type":"int256"},{"internalType":"uint256","name":"openNotional","type":"uint256"}],"stateMutability":"view","type":"function"},{"inputs":[{"internalType":"uint256","name":"_num","type":"uint256"}],"name":"setNumAMMs","outputs":[],"stateMutability":"nonpayable","type":"function"},{"inputs":[],"name":"settleFunding","outputs":[],"stateMutability":"nonpayable","type":"function"},{"inputs":[{"components":[{"internalType":"uint256","name":"ammIndex","type":"uint256"},{"internalType":"address","name":"trader","type":"address"},{"internalType":"int256","name":"baseAssetQuantity","type":"int256"},{"internalType":"uint256","name":"price","type":"uint256"},{"internalType":"uint256","name":"salt","type":"uint256"}],"internalType":"struct IOrderBook.Order","name":"order","type":"tuple"},{"internalType":"bytes","name":"signature","type":"bytes"}],"name":"verifySigner","outputs":[{"internalType":"address","name":"","type":"address"},{"internalType":"bytes32","name":"","type":"bytes32"}],"stateMutability":"view","type":"function"}]}`
```
[the remark this answers is not in the document]
Comment (shubhamgoyal42) [reply plugin/evm/orderbook_test.go:24]: no, cuz OrderBookAbi in abis.go doesn't have methods, only events are mentioned there

File: contract-examples/contracts/hubble-v2/OrderBook.sol [hunk 4]
```diff
@@ -50,31 +50,32 @@ contract OrderBook is IOrderBook, EIP712Upgradeable {
...
-            fulfillPrice = orders[1].price;
-        }
+        // if (orders[0].price != orders[1].price && blockPlaced0 > blockPlaced1) {
+        //     fulfillPrice = orders[1].price;
+        // }
 
-        _openPosition(orders[0], fillAmount, fulfillPrice);
-        _openPosition(orders[1], -fillAmount, fulfillPrice);
+        // _openPosition(orders[0], fillAmount, fulfillPrice);
+        // _openPosition(orders[1], -fillAmount, fulfillPrice);
```
[the remark this answers is not in the document]
Comment (shubhamgoyal42) [reply contract-examples/contracts/hubble-v2/OrderBook.sol:76]: i'll try with keeping this logic

[the remark this answers is not in the document]
Comment (shubhamgoyal42) [reply contract-examples/contracts/hubble-v2/OrderBook.sol]: Added this logic, it worked even with this logic

File: plugin/evm/orderbook_test.go (hunk 1 above)
Comment (shubhamgoyal42) [reply plugin/evm/orderbook_test.go]: Mistake in the comments. Reorg happens when vm1 accepts block C. Updated the comments now

File: plugin/evm/orderbook_test.go (hunk 2 above)
Comment (shubhamgoyal42) [reply plugin/evm/orderbook_test.go:135]: Asserting that order 2(bob's short order) exists in memory for both VMs

**Review by atvanguard — APPROVED** [review]

**Pull request merged**